crates/bevy_input/src/mouse.rs

```diff
@@ -2,46 +2,95 @@ use crate::{ElementState, Input};
 use bevy_ecs::{event::EventReader, system::ResMut};
 use bevy_math::Vec2;
 
-/// A mouse button input event
+/// A mouse button input event.
+///
+/// This event is the translated version of the `WindowEvent::MouseInput` from the `winit` crate.
+/// It is available to the end user and can be used for game logic.
+///
+/// ## Usage
+///
+/// The event is read inside of the [`mouse_button_input_system`](crate::mouse::mouse_button_input_system)
+/// to update the [`Input<MouseButton>`](crate::Input<MouseButton>) resource.
 #[derive(Debug, Clone)]
 pub struct MouseButtonInput {
+    /// The mouse button assigned to the event.
     pub button: MouseButton,
+    /// The pressed state of the button.
     pub state: ElementState,
 }
 
-/// A button on a mouse device
+/// A button on a mouse device.
+///
+/// ## Usage
+///
+/// It is used as the generic `T` value of an [`Input`](crate::Input) to create a `bevy`
+/// resource. The resource stores the data of the buttons of a mouse and can be accessed
+/// inside of a system.
```

Comment:
I don't understand this bit: "The resource stores the data of the buttons of a mouse"
```suggestion
/// resource. It can be accessed inside of a system with a `Res<Input<MouseButton>>`
/// parameter.
```
(Second half is just a suggestion. It's only the "data of a button" that I think should be really fixed)